### PR TITLE
DM-14291: PolyMap.polyTran does not clear IterInverse

### PR DESCRIPTION
**Pull request by r-owen**

### CODE REVIEW

File: polymap.c [hunk 1]
```diff
@@ -1957,20 +1974,23 @@ static void IterInverse( AstPolyMap *this, AstPointSet *out, AstPointSet *result
...
 
 *  Description:
-*     This function transforms a set of PolyMap output positions using
-*     the inverse transformation of the PolyMap, to generate the corresponding
-*     input positions. An iterative Newton-Raphson method is used which
-*     only required the forward transformation of the PolyMap to be deifned.
+*     This function transforms a set of PolyMap positions using the original
+*     inverse transformation of the PolyMap (i.e. the Negated attribute
+*     is assumed to be zero). An iterative Newton-Raphson method is used
+*     which only required the original forward transformation of the PolyMap
```
Comment (natelust) [inline polymap.c:1980]: requires  not required

File: polymap.c [hunk 2]
```diff
@@ -1957,20 +1974,23 @@ static void IterInverse( AstPolyMap *this, AstPointSet *out, AstPointSet *result
...
+*     to be defined.
 
 *  Parameters:
 *     this
 *        The PolyMap.
 *     out
-*        A PointSet holding the PolyMap output positions that are to be
-*        transformed using the inverse transformation.
+*        A PointSet holding the positions that are to be transformed using
+*        the original inverse transformation. These corresponds to
```
Comment (natelust) [inline polymap.c:1988]: corresponds -> correspond

File: polymap.c [hunk 3]
```diff
@@ -1957,20 +1974,23 @@ static void IterInverse( AstPolyMap *this, AstPointSet *out, AstPointSet *result
...
+*        outputs of the original (i.e. uninverted) PolyMap
 *     result
-*        A PointSet into which the generated PolyMap input positions are to be
-*        stored.
+*        A PointSet into which the transformed positions are to be stored.
+*        These corresponds to inputs of the original (i.e. uninverted) PolyMap
```
Comment (natelust) [inline polymap.c:1992]: corresponds -> correspond

File: polymap.c [hunk 4]
```diff
@@ -3709,7 +3736,14 @@ f     function is invoked with STATUS set to an error value, or if it
 
 /* If an error occurred, or the fit was not good enough, annul the returned
    PolyMap. */
-   if ( !ok || !astOK ) result = astAnnul( result );
+   if ( !ok || !astOK ) {
+      result = astAnnul( result );
+
```
Comment (natelust) [inline polymap.c:3741]: This space seems weird to me. I feel like if you are going to have a comment above the else, then it is acceptable to end the } on a line, put the comment in, and have the else. Or have the comment in the else block above the actual work. I don't have a strong feeling if you want to leave it. Just thought I would mention that the space makes for awkward reading, and seems to be an unmatched {} while reading.

**Review by natelust — APPROVED** [review]

File: polymap.c (hunk 4 above)
Comment (r-owen) [reply polymap.c:3741]: David Berry has his own standards for formatting C code which put spaces where LSST does not.

The curly braces are correctly matched, fortunately. If you read down a few lines you'll see that the next code is `} else {` preceded by an explanatory comment.

**Pull request closed without merging**